src/Hyper/Node/Server.purs

```diff
@@ -82,30 +84,36 @@ instance bufferNodeResponse :: (MonadAff e m)
   toResponse buf =
     ipure (write buf)
 
-readBody
+readBodyAsBuffer
   :: forall e.
      HttpRequest
-  -> Aff (http :: HTTP, avar :: AVAR | e) String
-readBody (HttpRequest request _) = do
+  -> Aff (http :: HTTP, avar :: AVAR, buffer :: BUFFER | e) Buffer
+readBodyAsBuffer (HttpRequest request _) = do
   let stream = HTTP.requestAsStream request
   completeBody <- makeVar
-  chunks <- makeVar' ""
-  e <- liftEff (catchException (pure <<< Just) (fillBody stream chunks completeBody *> pure Nothing))
-  case e of
-    Just err -> throwError err
-    Nothing -> takeVar completeBody
+  chunks <- makeVar' []
+  res <- liftEff $
+    catchException (pure <<< Left) (Right <$> fillBody stream chunks completeBody)
+  either throwError (const (takeVar completeBody)) res
   where
     fillBody stream chunks completeBody = do
```

Comment:
This part catches the `EXCEPTION` effect from `Stream.onData` (which seems to be thrown if the Node stream emits Chunk that is not a String or a Buffer). How that relates to `Stream.onError`, I'm not sure, but we should perhaps handle the `onError` case as well.